### PR TITLE
Get providers

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java [hunk 1]
```diff
@@ -144,4 +145,18 @@ public ResponseEntity<Double> getDefaultResultsThreshold() {
         return ResponseEntity.ok().body(service.getDefaultResultThresholdValue());
     }
 
+    @GetMapping(value = "/providers", produces = APPLICATION_JSON_VALUE)
+    @Operation(summary = "Get all sensitivity analysis providers")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Sensitivity analysis providers have not been found")})
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java:150]: Sensitivity analysis providers have been found

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java [hunk 2]
```diff
@@ -144,4 +145,18 @@ public ResponseEntity<Double> getDefaultResultsThreshold() {
...
+    public ResponseEntity<List<String>> getProviders() {
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON)
+                .body(service.getProviders());
+    }
+
+    @GetMapping(value = "/default-provider", produces = TEXT_PLAIN_VALUE)
+    @Operation(summary = "Get sensitivity analysis default provider")
+    @ApiResponses(@ApiResponse(responseCode = "200", description = "The sensitivity analysis default provider has not been found"))
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java:158]: The sensitivity analysis default provider has been found

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java]: Done

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java]: Done

**Review by bendaamerahm — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**